src/strategy.rs

```diff
@@ -40,1645 +33,159 @@ impl Clone for Box<dyn Evaluator + Send + Sync + 'static> {
     }
 }
 
-/// <https://docs.getunleash.io/user_guide/activation_strategy#standard>
-pub fn default<S: BuildHasher>(_: Option<HashMap<String, String, S>>) -> Evaluate {
-    Box::new(|_: &Context| -> bool { true })
-}
+type StrategyDefinitions = HashMap<String, Vec<(String, Option<HashMap<String, String>>)>>;
 
-/// <https://docs.getunleash.io/user_guide/activation_strategy#userids>
-/// userIds: user,ids,to,match
-pub fn user_with_id<S: BuildHasher>(parameters: Option<HashMap<String, String, S>>) -> Evaluate {
-    let mut uids: HashSet<String> = HashSet::new();
-    if let Some(parameters) = parameters {
-        if let Some(uids_list) = parameters.get("userIds") {
-            for uid in uids_list.split(',') {
-                uids.insert(uid.trim().into());
-            }
-        }
-    }
-    Box::new(move |context: &Context| -> bool {
-        context
-            .user_id
-            .as_ref()
-            .map(|uid| uids.contains(uid))
-            .unwrap_or(false)
-    })
+pub struct CustomStrategyHandler {
```

Comment:
This change is hard to read because Github is being daft here but essentially this wipes out all the contents of this file and adds a custom strategy handler, similar to the handlers in Yggdrasil's host FFI layers ([ruby example](https://github.com/Unleash/yggdrasil/blob/main/ruby-engine/lib/custom_strategy.rb).)